### ISSUE TITLE
npm install sub-dependencies problem.

### ISSUE THREAD

**Issue by dfcreative** [npm/npm#11256]
Hello!

With the npm@3+ I started getting troubles with `npm install <module>` and `npm update <module>`.

Suppose a dep structure defined in `package.json` like that:
```
a@1.0.0                  ← we’re working on this guy
├── b@1.0.0
└── c@1.0.0
    └── b@1.0.0
```

First `npm install` is correct.
Later though, if I update `c@1.0.0` to `c@1.0.1`, and do `npm update c` or `npm install c`, I just loose `b` module from my `a` package (`b` migrates to `c` i guess), so I start getting an error `Cannot find module 'b'` and I’m forced to do `npm install` of everything once again. Quite annoying.

Is there something I misunderstand or an issue I should’ve found and read about that?

Win7, npm@3.3.12, node@5.3.0

Thank you.

**Comment by felixfbecker** [comment]
If a depends on b directly, it should state that in its package.json

**Comment by dfcreative** [comment]
Yes, that is what "dep structure" means. And it works as described.

**Comment by rioc0719** [comment]
I get the same error. My package depends on Gulp and chalk. One of Gulp's dependencies, liftoff, depends on chalk also. When I install Gulp at the top level of my project, chalk gets uninstalled (even though my project states that it depends on chalk in its package.json) and I have to manually reinstall it. This happens almost any time my project and one of its deps depend on a popular module.

**Comment by dfcreative** [comment]
@rioc0719 try updating npm, version 3.7.0 works fine. There are other bugs, but less annoying.

**Comment by ashleygwilliams** [comment]
sounds like this is solved by `3.7.0`. closing. please open another issue if you continue to have problems!
